documentation/Software Requirements Specification.md

```diff
@@ -0,0 +1,331 @@
+* Price – the total price of the car;
+* Add date – the date the order was added;
+
+
+***Filtering by date:***
+
+* In the order list view mode, the user sets a date filter and presses the refresh list button (to the right of the date entry field); 
+* The application will display a form to view the list of orders with updated data. 
+### 1.2 Add order
+
+***Main scenario:***
+* User clicks the “Add” button in the orders list view mode;
+* Application displays form to enter order data;
+* User enters order data and presses “Save” button;
+* If any data is entered incorrectly, incorrect data messages are displayed;
+* If entered data is valid, then record is adding to database;
+* If error occurs, then error message is displaying;
+* If new order record is successfully added, then list of orders with added records is displaying.
+
+***Cancel operation scenario:***
+* User clicks the “Add” button in the order list view mode;
+* Application displays form to enter order data;
+* User enters order data and presses “Cancel” button;
+* Data don’t save in data base, then list of orders records is displaying to user;
+* If the user selects the menu item "Orders”, ”Clients” or "Cars", the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+![](add_order.PNG)
+<br>Pic. 1.2 Add order.
+
+***When adding a order, the following details are entered:***
+* Client  – client’s first and last name;
+* Car  – car description;
+* Price – the total price of the car;
+* Add date – date of adding order;
+
+### 1.3 Edit order.
+
+***Main scenario:***
+* User clicks the “Edit” button in the orders list view mode; 
+* Application displays form to enter order data; 
+* User enters order data and presses “Save” button; 
+* If any data is entered incorrectly, incorrect data messages are displayed; 
+* If entered data is valid, then edited data is added to database; 
+* If error occurs, then error message is displaying; 
+* If order record is successfully edited, then list of orders with added records is displaying. 
+
+***Cancel operation scenario:***
+* User clicks the “Edit” button in the orders list view mode; 
+* Application displays form to enter order data; 
+* User enters order data and presses “Cancel” button; 
+* Data don’t save in data base, then list of orders records is displaying to user. 
+* If the user selects the menu item "Orders”, ”Clients” or "Cars", the data will not be saved to the database and the corresponding form with updated data will be opened.
+
+![](edit_order.PNG)
+<br>Pic. 1.3 Edit order.
+
+***When editing a order, the following details are entered:***
+* Client  – client’s first and last name;
+* Car  – car description;
+* Price – the total price of the car;
+* Add date – date of adding order;
+
+***Constraints for data validation:***
+* Client  – maximum length of 90 characters; 
+* Car  – maximum length of 90 characters;
+* Price – maximum length of 90 characters;
```

Comment:
Price validation by length is not a good way here. This field should be a simple numeric field not a string